src/runtime_src/core/pcie/noop/pcidev_noop.h

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+#ifndef _XCL_PCIDEV_NOOP_H_
+#define _XCL_PCIDEV_NOOP_H_
+
+#include "device_noop.h"
+#include "core/common/device_factory.h"
+#include <string>
+
+namespace xrt_core { namespace pci {
+
+class pcidev_noop : public device_factory
```

Comment:
This is not a  pcidev, it is a device factory.  Naming is important and `pcidev` is just not good. Please call it what it is.
This would work
`class device_factory : public xrt_core::device_factory`
but maybe you'd prefer 
`class device_factory_noop : public device_factory`